### PR TITLE
Baltcogo data provider

### PR DESCRIPTION
**Pull request by martypowell**
Added a data provider for easy of use when configuring calls to the data that provides the dropdown values.

This makes it easier switching between development and prod, but wonder if we could make this even easier in the future.

### CODE REVIEW

File: js/page-specific/baltcogo-app/providers/data-provider.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// http://www.learn-angular.org/#!/lessons/the-provider-recipe
+(function (app, querystringer) {
```
Comment (mxsnyder) [inline js/page-specific/baltcogo-app/providers/data-provider.js]: It looks like you're not using querystringer, and can safely remove it.

File: js/page-specific/baltcogo-app/BaltCoGoReporterCtrl.js [hunk 2]
```diff
@@ -1,504 +1,530 @@
...
-
-			if (keyCode === 13) {
-				event.preventDefault();
-			}		
-		}
-
-	}
+  'use strict';
+
+  function Data(arr) {
```
Comment (mxsnyder) [inline js/page-specific/baltcogo-app/BaltCoGoReporterCtrl.js]: Why are the Data function and the provider here? You have them elsewhere, so just inject it from the other file.

File: js/page-specific/baltcogo-app/BaltCoGoReporterCtrl.js [hunk 3]
```diff
@@ -1,504 +1,530 @@
...
+      urlArr = keyValArr;
+    };
+
+    this.$get = [function () {
+      return new Data(urlArr);
+    }];
+  });
+
+
+  app.controller('BaltCoGoReporterCtrl', ['$http', '$scope', '$timeout', 'mapServiceComposite', 'reportService', "data", reporterController]);
```
Comment (mxsnyder) [inline js/page-specific/baltcogo-app/BaltCoGoReporterCtrl.js]: "data" should be in single quotes, but see my comment below about giving it a better name.

File: js/page-specific/baltcogo-app/providers/data-provider.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    for (var i = 0; i < arr.length; i++) {
+      var obj = arr[i];
+      var key = Object.keys(obj)[0];
+
+      this.endpoints[key] = obj[key];
+    }
+  }
+
+  app.provider('data', function () {
```
Comment (mxsnyder) [inline js/page-specific/baltcogo-app/providers/data-provider.js]: This needs a better name. urlProvider maybe?

File: js/page-specific/baltcogo-app/providers/data-provider.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    var urlArr = null;
+
+    this.setUrls = function (keyValArr) {
```
Comment (mxsnyder) [inline js/page-specific/baltcogo-app/providers/data-provider.js]: Elsewhere in the code, we follow the **var foo = function () { ... }; return { foo: foo };** pattern. I see the original author likes using _this_ all over the place, but I think should should still be adapted into our style.

File: js/page-specific/baltcogo-app/providers/data-provider.js (hunk 1 above)
Comment (martypowell) [inline js/page-specific/baltcogo-app/providers/data-provider.js]: removed.

File: dist/js/baltcogo-reporter.js [hunk 6]
```diff
@@ -307,45 +309,20 @@
...
-    };
-
-    this.$get = [function () {
-      return new Data(urlArr);
-    }];
-  });
-
-
-  app.controller('BaltCoGoReporterCtrl', ['$http', '$scope', '$timeout', 'mapServiceComposite', 'reportService', "data", reporterController]);
+  app.controller('BaltCoGoReporterCtrl', ['$http', '$scope', '$timeout', 'mapServiceComposite', 'reportService', "url", reporterController]);
```
Comment (mxsnyder) [inline dist/js/baltcogo-reporter.js]: Change double quotes to single quotes.

File: dist/js/baltcogo-reporter.js [hunk 7]
```diff
@@ -2,46 +2,48 @@
...
       var obj = arr[i];
-      console.log(obj);
       var key = Object.keys(obj)[0];
 
       this.endpoints[key] = obj[key];
     }
   }
 
-  app.provider('data', function () {
+  app.provider('url', function () {
```
Comment (mxsnyder) [inline dist/js/baltcogo-reporter.js]: Naming this **url** is still confusing. You're calling it urlProvider locally in the controller once it's injected, so why not call it urlProvider here?

**Comment by martypowell** [comment]
for now i updated the provider name to be "urls" as we talked about, i couldn't figure out how to use "provider" in the name.

**Comment by mxsnyder** [comment]
The URLs will need to be updated in the configuration before this gets merged. We can hold it for 1.7.

**Comment by martypowell** [comment]
@mxsnyder closing this, as we are controlling these urls through a constants file now.

**Pull request closed without merging**